prusti-viper/src/encoder/procedure_encoder.rs

```diff
@@ -6028,12 +6028,16 @@ impl<'p, 'v: 'p, 'tcx: 'v> ProcedureEncoder<'p, 'v, 'tcx> {
         );
         match op {
             mir::NullOp::SizeOf => {
-                // TODO: ParamEnv::empty() should probably be tyctxt.param_env(def_id_of_method)
-                let bytes = self.encoder.env().tcx()
-                    .layout_of(ParamEnv::empty().and(op_ty))
-                    .unwrap().layout
-                    .size()
-                    .bytes();
+                let bytes = match self.encoder.env().tcx()
+                    .layout_of(self.encoder.env().query.tcx().param_env(self.proc_def_id).and(op_ty))
+                {
+                    Ok(layout_of) => layout_of.layout.size().bytes(),
+                    Err(_) => return Err(SpannedEncodingError::unsupported(
```

Comment:
```suggestion
                    Err(_) => return Err(SpannedEncodingError::internal(
```

`SpannedEncodingError::unsupported` messages are meant to be readable by regular Rust developers. Unless we can describe exactly what is unsupported, `SpannedEncodingError::internal` is more appropriate (and the error can be as ugly as we want).